src/Shared/FileMatcher.cs

```diff
@@ -2603,5 +2690,26 @@ private static bool DirectoryEndsWithPattern(string directoryPath, string patter
         };
 
         internal static bool IsRecursiveDirectoryMatch(string path) => path.TrimTrailingSlashes() == recursiveDirectoryMatch;
+
+        public struct GetFilesImplementationResults
+        {
+            public static readonly GetFilesImplementationResults Default = new GetFilesImplementationResults(Array.Empty<string>(), SearchAction.None, string.Empty, false);
+
+            public GetFilesImplementationResults(string[] fileList, SearchAction action, string fileSpec, bool isDriveEnumeratingExcludePattern)
+            {
+                FileList = fileList;
+                Action = action;
+                FileSpec = fileSpec;
+                IsDriveEnumeratingExcludePattern = isDriveEnumeratingExcludePattern;
+            }
+
+            public string[] FileList { get; set; }
+
+            public SearchAction Action { get; set; }
```

Comment:
I think I'm getting hung up on a naming thing: is `SearchAction` here really the _result_ of an enumeration? Like we do something and it's part of the results of that thing, indicating the combination of the state of the disk and the pattern?